File: java-semantickernel/src/main/java/com/codeyou/agent/StringPlugin.java
package com.codeyou.agent;

import com.microsoft.semantickernel.semanticfunctions.annotations.DefineKernelFunction;
import com.microsoft.semantickernel.semanticfunctions.annotations.KernelFunctionParameter;

/**
 * Plugin for string manipulation
 */
public class StringPlugin {
    
    @DefineKernelFunction(
        name = "reverseString",
        description = "Reverses a string",
        returnDescription = "The reversed string"
    )
    public String reverseString(
            @KernelFunctionParameter(
                name = "input",
                description = "The string to reverse"
            ) String input) {
        return new StringBuilder(input).reverse().toString();
    }
}

File: dotnet-semantickernel/StringPlugin.cs
using System.ComponentModel;
using Microsoft.SemanticKernel;

namespace SemanticKernelAgent
{
  /// <summary>
  /// Plugin for string manipulation
  /// </summary>
  public class StringPlugin
    {
        [KernelFunction, Description("Reverses a string")]
        public string ReverseString([Description("The string to reverse")] string input)
        {
            var charArray = input.ToCharArray();
            Array.Reverse(charArray);
            return new string(charArray);
        }
    }
}
